python/calico/felix/profilerules.py

```diff
@@ -149,6 +156,11 @@ def on_profile_update(self, profile, force_reprogram=False):
         assert not self._dead, "Shouldn't receive updates after we're dead."
         self._pending_profile = profile
         self._dirty |= force_reprogram
+        if profile:
+            if self._untracked is None:
+                self._untracked = profile['untracked']
+            else:
+                assert profile['untracked'] == self._untracked
```

Comment:
Yes, but I did try hard not to hide this: https://github.com/neiljerram/felix/blob/untracked/python/calico/felix/profilerules.py#L135

Anyway, I was thinking of it as an initial simplification that we could later remove, and it sounds like it might disappear as I address some of your other points.